lib/cn/vblock_builder.c

```diff
@@ -234,20 +337,22 @@ vbb_add_entry(
     assert(vlen);
     assert(vlen <= HSE_KVS_VALUE_LEN_MAX);
 
-    if (HSE_UNLIKELY(!_vblock_has_room(bld, vlen))) {
-        err = _vblock_finish(bld);
+    bld->cur_kobj = kobj;
+
+    if (HSE_UNLIKELY(!vblock_has_room(bld, vlen))) {
+        err = vblock_finish(bld);
         if (ev(err))
             return err;
     }
 
     if (HSE_UNLIKELY(!bld->blkid)) {
-        err = _vblock_start(bld);
+        err = vblock_start(bld);
         if (ev(err))
             return err;
     }
 
     assert(bld->blkid);
-    assert(_vblock_has_room(bld, vlen));
+    assert(vblock_has_room(bld, vlen));
```

Comment:
Yeah, guess it is making sure that the newly allocated vblock has room for vlen. That reminds me that I need to bump the min. supported mblock size from 1MiB to 2MiB so that it can fit a 1MiB value + a 4K footer.

Reply:
Right, but it already made sure of that, could it have changed between the live check and the assert?  I hope no!

Reply:
The live check and the assert could be on two different vblocks, i.e., the live check happens on a vblock that was finished above (in vblock_finish()) and the assert that follows happens on a newly allocated vblock (in vblock_start()).

Reply:
Ahhh, that wasn't clear to me...  So maybe that assert should be in the same block as vblock_start(), that might help make it more obvious...